java/java.mx.project/src/org/netbeans/modules/java/mx/project/SuiteActionProvider.java

```diff
@@ -139,27 +141,50 @@ public void invokeAction(String action, Lookup context) throws IllegalArgumentEx
                     return;
                 }
                 ListeningDICookie ldic = ListeningDICookie.create(-1);
-                Object obj = ldic.getArgs().get("port"); // NOI18N
                 DebuggerInfo di = DebuggerInfo.create(ListeningDICookie.ID, ldic);
-                DebuggerEngine[] engines = { null };
-                RequestProcessor.getDefault().post(() -> {
-                    DebuggerEngine[] engs = DebuggerManager.getDebuggerManager().startDebugging(di);
-                    engines[0] = engs[0];
+                ASYNC.post(() -> {
+                    DebuggerManager.getDebuggerManager().startDebugging(di);
                 });
                 int port = ldic.getPortNumber();
-                runMx(Bundle.MSG_Unittest(fo.getName()), "--attach", "" + port, "unittest", fo.getName() + testSuffix); // NOI18N
+                running = runBuildAndMx(null, Bundle.MSG_Unittest(fo.getName()), "--attach", "" + port, "unittest", fo.getName() + testSuffix); // NOI18N
                 break;
             default:
                 throw new UnsupportedOperationException(action);
         }
+        if (running != null) {
+            ActionProgress progress = ActionProgress.start(context);
+            running.handle((exitCode, error) -> {
+                progress.finished(exitCode == 0 && error == null);
+                return null;
+            });
+        }
     }
 
-    private boolean runMx(String taskName, String... args) {
+    private CompletionStage<Integer> ensureBuilt(FileObject fo) {
+        if (fo != null) {
+            SuiteSources.Group grp = prj.getSources().findGroup(fo);
+            if (grp != null) {
+                final String name = grp.getDisplayName();
+                return runMx(Bundle.MSG_BuildOnly(prj.getName(), name), "build", "--only", name); // NOI18N
```

Comment:
Should it be `.getDisplayName()` or `.getName()` for the `--only` argument ?

Reply:
`prj.getDisplayName()` doesn't exist. the `name` isn't a project (e.g. suite in mx terms) name, but rather a name of the source group - e.g. project in mx terms.

Reply:
Sorry, wrong wording. I wanted to ask: should it be
`..., "build", "--only", name);` or `..., "build", "--only", grp.getName());` ? Since `displayName` can be sometimes `displayName = name + "[" + rel + "]";`